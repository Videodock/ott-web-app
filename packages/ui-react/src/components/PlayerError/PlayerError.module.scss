@use '@jwp/ott-ui-react/src/styles/variables';
@use '@jwp/ott-ui-react/src/styles/theme';
@use '@jwp/ott-ui-react/src/styles/mixins/responsive';

.error {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: variables.$base-spacing;
}

.title {
  color: variables.$white;
  font-weight: var(--body-font-weight-bold);
  font-size: 34px;

  @include responsive.mobile-only() {
    font-size: 24px;
  }
}
