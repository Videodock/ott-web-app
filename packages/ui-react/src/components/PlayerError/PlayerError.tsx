import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './PlayerError.module.scss';

export enum PlayerErrorState {
  GEO_BLOCKED = 'GEO_BLOCKED',
}

type Props = {
  error: keyof typeof PlayerErrorState;
};

const PlayerError: React.FC<Props> = () => {
  const { t } = useTranslation('video');
  return (
    <div className={styles.error}>
      <h2 className={styles.title}>{t('player_error.geo_blocked_title')}</h2>
      <p>{t('player_error.geo_blocked_description')}</p>
    </div>
  );
};

export default PlayerError;
