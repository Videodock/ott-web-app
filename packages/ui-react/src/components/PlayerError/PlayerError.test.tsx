import React from 'react';
import { render } from '@testing-library/react';

import PlayerError, { PlayerErrorState } from './PlayerError';

describe('<ErrorPage>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<PlayerError error={PlayerErrorState.GEO_BLOCKED} />);

    expect(container).toMatchSnapshot();
  });
});
