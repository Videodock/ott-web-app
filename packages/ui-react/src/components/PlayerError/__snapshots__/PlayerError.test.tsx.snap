// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ErrorPage> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_error_41dece"
  >
    <h2
      class="_title_41dece"
    >
      player_error.geo_blocked_title
    </h2>
    <p>
      player_error.geo_blocked_description
    </p>
  </div>
</div>
`;
